centreon/src/Core/Common/Infrastructure/Routing/ModuleRouteLoader.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * Copyright 2005 - 2025 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Common\Infrastructure\Routing;
+
+use Core\Module\Infrastructure\ModuleVersionChecker;
+use Symfony\Bundle\FrameworkBundle\Routing\RouteLoaderInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\Routing\Loader\AttributeFileLoader;
+use Symfony\Component\Routing\RouteCollection;
+
+abstract readonly class ModuleRouteLoader implements RouteLoaderInterface
+{
+    public function __construct(
+        #[Autowire(service: 'routing.loader.attribute.file')] private AttributeFileLoader $loader,
+        #[Autowire(param: 'kernel.project_dir')] private string $projectDir,
```

Comment:
Please run CS fixer to check the impact on DI attributes.
I would have probably not chosen to inline them

Reply:
PHP CS Fixer seems ok with inlining attributes, but if it is a better convetion to put them a line above the parameters I'll move them. 
I personnaly like to have things inline if the line is < 120 char